xonsh/brace_expansion.py

```diff
@@ -0,0 +1,93 @@
+import glob
+import itertools
+import re
+
+# regex to be used for splitting on braced regions
+BRACED = re.compile(r"(?<!\\)(\{.*?(?<!\\)\})")
```

Comment:
This should probably be a lazyobject, see some of the other examples in the code base.